### PR TITLE
[SC-377] Decouple executor

### PR DESCRIPTION
**Pull request by hexonaut**
Splitting out business logic of executor from the bridge so that we can support multiple bridges in the future. The coverage needs to be improved as well. Can do in a separate PR.

### CODE REVIEW

File: src/executors/BridgeExecutorBase.sol [hunk 1]
```diff
@@ -405,4 +405,4 @@ abstract contract BridgeExecutorBase is IExecutorBase {
             }
         }
     }
-}
+}
```
Comment (lucas-manuel) [inline src/executors/BridgeExecutorBase.sol]: Revert this

File: src/interfaces/IExecutorBase.sol [hunk 2]
```diff
@@ -247,4 +247,4 @@ interface IExecutorBase {
      * @return True if the underlying action of actionHash is queued, false otherwise
      **/
     function isActionQueued(bytes32 actionHash) external view returns (bool);
-}
+}
```
Comment (lucas-manuel) [inline src/interfaces/IExecutorBase.sol]: Revert

File: src/receivers/BridgeExecutorReceiverArbitrum.sol [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        executor.queue(
+            targets,
+            values,
+            signatures,
+            calldatas,
+            withDelegatecalls
+        );
+    }
+
+}
```
Comment (lucas-manuel) [inline src/receivers/BridgeExecutorReceiverArbitrum.sol]: New line at end of file (Can make this a setting in settings.json so it does auto on save

File: src/interfaces/IExecutorBase.sol (hunk 2 above)
Comment (lucas-manuel) [inline src/interfaces/IExecutorBase.sol]: New line at end of file (Can make this a setting in settings.json so it does auto on save

File: src/interfaces/IAuthBridgeExecutor.sol [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+     **/
+    function queue(
+        address[] memory targets,
+        uint256[] memory values,
+        string[] memory signatures,
+        bytes[] memory calldatas,
+        bool[] memory withDelegatecalls
+    ) external;
+    
+}
```
Comment (lucas-manuel) [inline src/interfaces/IAuthBridgeExecutor.sol]: New line at end of file (Can make this a setting in settings.json so it does auto on save

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Looks good to me, want to add more testing but this looks solid

File: src/executors/BridgeExecutorBase.sol (hunk 1 above)
Comment (hexonaut) [reply src/executors/BridgeExecutorBase.sol]: done

File: src/interfaces/IExecutorBase.sol (hunk 2 above)
Comment (hexonaut) [reply src/interfaces/IExecutorBase.sol]: done

File: src/receivers/BridgeExecutorReceiverArbitrum.sol (hunk 3 above)
Comment (hexonaut) [reply src/receivers/BridgeExecutorReceiverArbitrum.sol]: done

File: src/interfaces/IAuthBridgeExecutor.sol (hunk 4 above)
Comment (hexonaut) [reply src/interfaces/IAuthBridgeExecutor.sol]: done

File: src/interfaces/IExecutorBase.sol (hunk 2 above)
Comment (hexonaut) [reply src/interfaces/IExecutorBase.sol]: done

**Review by lucas-manuel — APPROVED** [review]
Approving, still need to get to full coverage here

**Review by barrutko — APPROVED** [review]

**Pull request merged**